functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -0,0 +1,278 @@
+http://www.sqlskills.com/blogs/glenn/category/dmv-queries/
+
+The most recent version of the diagnostic queries are included in the dbatools module. 
+But it is possible to download a newer set or a specific version to an alternative location and parse and run those scripts.
+It will run all or a selection of those scripts on one or multiple servers and return the result as a PowerShell Object
+
+.PARAMETER SqlInstance
+The target SQL Server. Can be either a string or SMO server.
+	
+.PARAMETER Path
+Alternate path for the diagnostic scripts
+	
+.PARAMETER UseSelectionHelper
+Provides a gridview with all the queries to choose from and will run the selection made by the user on the Sql Server instance specified. 
+	
+.PARAMETER InstanceOnly
+Should maybe be Scope with the choices Instance and Database
+
+.PARAMETER DatabaseSpecific
+Specifies the path to the output files. 
+	
+.PARAMETER Silent
+Use this switch to disable any kind of verbose messages or progress bars
+
+.NOTES
+Author: André Kamman (@AndreKamman), http://clouddba.io
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.	
+
+.LINK
+https://dbatools.io/Invoke-DbaDiagnosticQuery
+
+
+.EXAMPLE   
+Invoke-DbaDiagnosticQuery -SqlInstance sql2016 -UseSelectionHelper
+
+Provides a gridview with all the queries to choose from and will run the selection made by the user on the Sql Server instance specified. 
+
+#>
+	
+	[CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+	param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[System.IO.FileInfo]$Path,
+		[string[]]$QueryName,
+		[switch]$UseSelectionHelper,
+		[switch]$InstanceOnly,
+		[switch]$DatabaseSpecific,
+		[switch]$Silent
+	)
+	
+	begin {
+		
+		function Invoke-DiagnosticQuerySelectionHelper {
+			Param (
+				[parameter(Mandatory = $true)]
+				$ParsedScript
+			)
+			
+			[string[]]$selection = $script | Select-Object QueryNr, QueryName, DBSpecific, Description | Out-GridView -Title "Diagnostic Query Overview" -OutputMode Multiple | Sort-Object QueryNr | Select-Object -ExpandProperty QueryName
```

Comment:
Should use `$ParsedScript` rather than `$script`. `$script` is from the calling scope and violating the scope boundary is _*evil*_